src/main/java/org/deepsymmetry/Wayang.java

```diff
@@ -73,11 +73,22 @@
     private static BufferedImage displayImage = null;
 
     /**
+     * Transfer object for asynchronous calls.
+     */
+    private static Transfer transfer = null;
+
+    /**
      * Close the Push 2 interface if it is open, and shut down our libusb context if it is active.
      */
     public static synchronized void close() {
         if (pushHandle != null) {
             displayImage = null;
+
+            // handle all remaining events
+            int result = LibUsb.handleEventsTimeout(null, 250000);
```

Comment:
Here is where we would abort and join the event processing thread if it was running (which it would only be if someone had called `sendFrameAsync()`), as described in the usb4java documentation.